### PR TITLE
refactored /daemonstatus

### PR DESCRIPTION
**Pull request by vlerkin**
refactored /daemonstatus to contain node_name, status and the number of pending, running and finished jobs, tested that it shows the actual load

### CODE REVIEW

File: scrapyd_k8s/api.py [hunk 1]
```diff
@@ -25,7 +25,23 @@ def healthz():
 
 @app.get("/daemonstatus.json")
 def api_daemonstatus():
-    return { "status": "ok", "spiders": 0 }
+    node_name = config.scrapyd().get("node_name", launcher.get_node_name())
```
Comment (wvengen) [inline scrapyd_k8s/api.py:28]: this can be moved to the return statement

File: scrapyd_k8s/api.py [hunk 2]
```diff
@@ -25,7 +25,23 @@ def healthz():
...
+    total_pending = 0
+    total_running = 0
+    total_finished = 0
+    projects = config.listprojects()
+    for project_id in projects:
```
Comment (wvengen) [inline scrapyd_k8s/api.py:33]: what about `for project_id in config.list_projects():`?

File: scrapyd_k8s/api.py [hunk 3]
```diff
@@ -25,7 +25,23 @@ def healthz():
...
+        jobs = launcher.listjobs(project_id)
```
Comment (wvengen) [inline scrapyd_k8s/api.py:34]: Why do you iterate over the projects? It would be much lighter to query all jobs in one go, without the project filter.

**Review by wvengen — COMMENTED** [review]
Ah, great, thank you!
This can be a little more performant and concise. Otherwise, good to go.

File: scrapyd_k8s/api.py (hunk 3 above)
Comment (vlerkin) [reply scrapyd_k8s/api.py:34]: Yeah, I thought that jobs can be divided across projects and if I won't check all the jobs of all the projects, I might miss something, I guess we can just simply access all the jobs

File: scrapyd_k8s/api.py [hunk 4]
```diff
@@ -25,7 +25,17 @@ def healthz():
 
 @app.get("/daemonstatus.json")
 def api_daemonstatus():
-    return { "status": "ok", "spiders": 0 }
+    jobs = launcher.listjobs()
+    total_pending = len([j for j in jobs if j['state'] == 'pending'])
+    total_running = len([j for j in jobs if j['state'] == 'running'])
+    total_finished = len([j for j in jobs if j['state'] == 'finished'])
```
Comment (wvengen) [inline scrapyd_k8s/api.py:31]: I think these can also move to the return statement, right?

**Review by wvengen — COMMENTED** [review]
Thanks! Almost there :)

File: scrapyd_k8s/api.py (hunk 4 above)
Comment (wvengen) [reply scrapyd_k8s/api.py]: It's just a matter of codestyle, terseness vs. smaller bits per line. Thanks for fixing!

**Comment by wvengen** [comment]
Not sure why the test is failing now, but would be good to look into.

**Comment by wvengen** [comment]
Restarted the test, and is fine. Hope this happens too often :+1:

**Comment by wvengen** [comment]
Ah, good one, about the iterator! Didn't realise it either when reading :o:

**Comment by wvengen** [comment]
Ah, please mention the issue number when opening a PR :) #23 here

(And as a sidenote, we could also make a list of just the state, not requiring the whole objects in memory; but I'm not sure if that improves the code (readability), I was happy to merge it as it is.)

**Pull request merged**